kvcoord: small fixes

andreimatei opened the pull request:
See individual commits.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46407)
<!-- Reviewable:end -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46407#-:-M36t-oj6A2CoyQRA9-k:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)

___
*[pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go, line 285 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/46407#-M36svVn7VzV31Q7Jad7:-M36svVn7VzV31Q7Jad8:b3rvybn) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7efcba03c4310d90522de84ae23fe7bcaa5e19e/pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go#L285)):*
> ```Go
> 	}
> 
> 	log.VEventf(ctx, 2, "retrying %s at refreshed timestamp %s because of %s",
> ```

Shouldn't we also move this down then, too?


<!-- Sent from Reviewable.io -->

andreimatei committed:
kvcoord: improve a log message

Release note: None

andreimatei committed:
roachpb: remove unused arg

Release note: None

andreimatei committed:
kvcoord: don't do scary ReadTimestamp bump

When seeing a WriteTooOld flag, the span refresher is generating a
synthetic retriable error, on which to refresh. The point of this error
is to mimic a WriteTooOldError returned by the server. However, we were
not mimicking very well; we were bumping the read timestamp on the txn
inside this error, which is not what the server does. Read timestamps
should only be updated after a successful refresh, not before attempting
one.
The read timestamp from the synthetic error was flowing into the
sr.refreshedTimestamp field in the case when the refresh was not
successful. This is not cool, since the refreshedTimestmap field is
supposed to track successful refreshes. I think in the end it's
inconsequential because, after a failed refresh, we're returning a
retriable error which eventually bumps the epoch, which in turn resets
the refreshedTimestamp field.

Release note: None

andreimatei reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46407)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @nvanbenschoten)

___
*[pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go, line 285 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/46407#-M36svVn7VzV31Q7Jad7:-MFGqjZo3AM9fzseJrCE:b-kmsypq) ([raw file](https://github.com/cockroachdb/cockroach/blob/b7efcba03c4310d90522de84ae23fe7bcaa5e19e/pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go#L285)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Shouldn't we also move this down then, too?
</blockquote></details>

changed this log statement


<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46407#-:-MFH4lJKErA_bQpG35dA:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @nvanbenschoten)



<!-- Sent from Reviewable.io -->
